pkg/sdk/roles_validations.go

```diff
@@ -0,0 +1,57 @@
+package sdk
+
+import "errors"
+
+func (opts *CreateRoleOptions) validateProp() error {
```

Comment:
And what about nil options? If we use DTOs we can enforce it, but now nil options here will result in panic

Reply:
added nil checks